posydon/unit_tests/utils/test_configfile.py

```diff
@@ -0,0 +1,290 @@
+            mock_node = AST()
+            mock_node.body = parse(source, mode='eval')
+            return mock_node
+        # missing argument
+        with raises(TypeError, match="missing 1 required positional"+\
+                    " argument: 'inifile'"):
+            totest.parse_inifile()
+        # bad input
+        with raises(TypeError, match="'int' object is not iterable"):
+            totest.parse_inifile(1)
+        # read test ini
+        rp, s, mi, me = totest.parse_inifile(test_path)
+        assert rp == {'MESA_DIR'.lower(): totest.os.environ['MESA_DIR'],\
+                      'test_bool1'.lower(): True, 'test_bool2'.lower(): False,\
+                      'test_None'.lower(): None}
+        assert s == {'MESA_DIR'.lower(): totest.os.environ['MESA_DIR'],\
+                     'test_exception'.lower():\
+                     ["ast.parse('X=1'", "mode='eval')"]}
+        assert mi == {'MESA_DIR'.lower(): totest.os.environ['MESA_DIR'],\
+                      'test_float'.lower(): 0.1, 'test_int'.lower(): 10,\
+                      'test_list'.lower(): ['Unit Test1', 'Unit Test2'],\
+                      'test_str1'.lower(): 'Unit Test',\
+                      'test_str2'.lower(): ['Unit', 'Test'],\
+                      'test_BinOp'.lower(): 11}
+        assert me == {'MESA_DIR'.lower(): totest.os.environ['MESA_DIR'],\
+                      'test_else'.lower(): "print('1')"}
+        with monkeypatch.context() as mp:
+            # replace parse to check iteration
+            mp.setattr(totest.ast, "parse", mock_parse)
+            # replace content of test.ini to check Error
+            with open(test_path, "w") as test_file:
+                test_file.write("[Unit Test]\n")
+                test_file.write("test_value = 'Test'\n")
+            with raises(KeyError, match="'run_parameters'"):
+                totest.parse_inifile(test_path)
+
+
+class TestConfigFile:
+    @fixture
+    def ConfigFile(self):
+        # initialize an instance of the class with defaults
+        ConfigFile = totest.ConfigFile()
+        return ConfigFile
+
+    @fixture
+    def test_path(self, tmp_path):
+        # a temporary path for testing
+        return totest.os.path.join(tmp_path, "test.json")
+
+    @fixture
+    def test_json(self, test_path):
+        # a temporary json file for testing
+        with open(test_path, "w") as test_file:
+            test_file.write('{\n    "Unit": "Test"\n}\n')
+        return
+
+    @fixture
+    def test_entries(self):
+        # entries for testing
+        return {'Unit1': "Test", 'Unit2': "Again"}
+
+    @fixture
+    def ConfigFile_filled(self, test_entries):
+        # initialize an instance of the class with defaults
+        ConfigFile = totest.ConfigFile()
+        ConfigFile.update(test_entries)
+        return ConfigFile
+
+    # test the ConfigFile class
+    def test_init(self, ConfigFile, tmp_path):
```

Comment:
I'm not aware whether there is a way to create an object in python without its `__init__` to be run. Hence, running it here explicitly would be the second execution of it.
Otherwise, I don't see any advantage of not using a fixture, while not questioning the usage of fixtures at all.

Reply:
I don't think I was clear. I meant that you call `__init__` inside `test_init` to test the initialisation, but this should be fine.